### PR TITLE
9479 Add Stamp Disposition

### PR DESCRIPTION
**Pull request by rosiehsmith**
# This PR is the work done so far on 9479 (the happy path) in order to give a preview of it at the demo tomorrow 7/28. 
## Related, smaller PRs to the story branch: 
- https://github.com/flexion/ef-cms/pull/9527
- https://github.com/flexion/ef-cms/pull/9529
- https://github.com/flexion/ef-cms/pull/9530
- https://github.com/flexion/ef-cms/pull/9531
- https://github.com/flexion/ef-cms/pull/9537
- https://github.com/flexion/ef-cms/pull/9541
- https://github.com/flexion/ef-cms/pull/9543
- https://github.com/flexion/ef-cms/pull/9546
- https://github.com/flexion/ef-cms/pull/9550
- https://github.com/flexion/ef-cms/pull/9551
- https://github.com/flexion/ef-cms/pull/9552

## Work that is still left for this to be complete:
- Generated Stamp Orders should appear in Order search results - this is likely not happening now because `generateStampedCoversheetInteractor` is missing `applicationContext.getDocumentGenerators().order` and subsequent calls to save the content of the pdf
- Message flow - starting on the messages page, then apply the stamp - return to messages instead of navigating to drafts. Find out if the most recent message should be selected rather than the first message associated with that attachment.
- Automated testing: `generateCoverSheetData`, `generateStampedCoversheetInteractor`
- Clear optional button should empty out the customText box
- Form clearing: if you set the date, then clear, and reset the same date field it doesn't work

### DOD:
  - PDF visual test with the generated stamped coversheet
  - Integration testing
    - Verify the behavior of the back button for Messages and Document Preview (see [comment](https://github.com/flexion/ef-cms/pull/9552#discussion_r931228665))
  - Consider combining endpoint calls in `completeStampingMotionAction` to increase performance
  - Consider if we need a `Stamp` entity just for validation. Since it's being passed through as part of the docket entry, maybe the entity is not needed.
  - Revisit height adjustment of `docket-entry-preview-text` to not be hardcoded - talk to UX
  - Does the back button on the stamp form need to have specific words for which page you came from?
  - Consider not sending `formattedDraftDocumentTitle` thru the proxy if possible
  - If there's logic in `ApplyStamp.jsx`, move it to a helper
  - Test rawDocketEntry.stampData defaults to {}
  - Validate Stamp before putting it on the DocketEntry
  - Find out from users if two served stamps on the stamp order is confusing (might just wait for user feedback) --  done as a part of feedback. Only one served label shows up now

### CODE REVIEW

File: shared/src/business/useCases/addDraftStampOrderDocketEntryInteractor.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+/**
+ * addDraftStampOrderDocketEntryInteractor
+ *
+ * @param {object} applicationContext the application context
+ * @param {object} providers the providers object
+ * @param {string} providers.docketNumber the docket number of the case on which to save the document
+ * @param {string} providers.formattedDraftDocumentTitle the formatted draft document title of the document
+ * @param {string} providers.originalDocketEntryId the id of the original (unsigned) document
+ * @param {string} providers.signedDocketEntryId the id of the signed document
```
Comment (rosiehsmith) [inline shared/src/business/useCases/addDraftStampOrderDocketEntryInteractor.js]: Rename this to stampedDocketEntryId

File: web-client/src/presenter/actions/setPDFForStampAction.test.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      numPages: 1,
+    };
+
+    removeCoverMock = jest.fn();
+    getPagesMock = jest.fn();
+
+    applicationContext
+      .getUseCases()
+      .loadPDFForSigningInteractor.mockImplementation(
+        (_applicationContext, { onlyCover }) => {
```
Comment (rosiehsmith) [inline web-client/src/presenter/actions/setPDFForStampAction.test.js]: _applicationContext - what is it?

File: web-client/src/presenter/actions/StampMotion/clearOptionalFieldsStampFormAction.test.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  it('should unset contact address info on the form', async () => {
+    const { state } = await runAction(clearOptionalFieldsStampFormAction, {
+      state: {
+        form: {
+          day: '30',
+          dueDateMessage: 'the end is nigh',
+          jurisdictionalOption: 'as moot',
+          month: '30',
+          strickenFromTrialSession: true,
+          year: '1999',
```
Comment (rosiehsmith) [inline web-client/src/presenter/actions/StampMotion/clearOptionalFieldsStampFormAction.test.js:15]: add customText to the state before clearing

File: web-client/src/presenter/actions/StampMotion/clearOptionalFieldsStampFormAction.test.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { clearOptionalFieldsStampFormAction } from '../StampMotion/clearOptionalFieldsStampFormAction';
+import { runAction } from 'cerebral/test';
+
+describe('clearOptionalFieldsStampFormAction', () => {
+  it('should unset contact address info on the form', async () => {
```
Comment (rosiehsmith) [inline web-client/src/presenter/actions/StampMotion/clearOptionalFieldsStampFormAction.test.js:5]: test name

File: web-client/src/presenter/actions/StampMotion/clearPDFStampDataAction.test.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import { clearPDFStampDataAction } from './clearPDFStampDataAction';
+import { runAction } from 'cerebral/test';
+
+describe('clearPDFStampDataAction', () => {
+  it('should unset state.pdfForSigning.stampData', async () => {
+    const { state } = await runAction(clearPDFStampDataAction, {
+      state: {
+        pdfForSigning: {
+          stampData: {
```
Comment (rosiehsmith) [inline web-client/src/presenter/actions/StampMotion/clearPDFStampDataAction.test.js]: revisit this as the stamp is no longer dynamically placed

File: web-client/src/presenter/actions/StampMotion/setSuccessfulStampFromDocumentTitleAction.test.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import { applicationContextForClient as applicationContext } from '../../../../../shared/src/business/test/createTestApplicationContext';
+import { presenter } from '../../presenter-mock';
+import { runAction } from 'cerebral/test';
+import { setSuccessfulStampFromDocumentTitleAction } from './setSuccessfulStampFromDocumentTitleAction';
+
+describe('setSuccessfulStampFromDocumentTitleAction,', () => {
+  presenter.providers.applicationContext = applicationContext;
+
+  it('sets the success message from the documentTitle when the document eventCode is not PSDE', async () => {
```
Comment (rosiehsmith) [inline web-client/src/presenter/actions/StampMotion/setSuccessfulStampFromDocumentTitleAction.test.js]: test name

File: web-client/src/presenter/actions/StampMotion/unsetDeniedOptionsOnStampFormAction.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import { state } from 'cerebral';
+
+/**
+ * unsets state.form denied option fields
+ *
+ * @param {object} providers the providers object
+ * @param {Function} providers.store the cerebral store function
```
Comment (rosiehsmith) [inline web-client/src/presenter/actions/StampMotion/unsetDeniedOptionsOnStampFormAction.js:8]: docs for g et

File: web-api/src/documents/generateStampedCoversheetLambda.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+ * @returns {Promise<*|undefined>} the api gateway response object containing the statusCode, body, and headers
+ */
+exports.generateStampedCoversheetLambda = event =>
+  genericHandler(
+    event,
+    async ({ applicationContext }) => {
+      await applicationContext
+        .getUseCases()
+        .generateStampedCoversheetInteractor(applicationContext, {
+          ...event.pathParameters,
```
Comment (codyseibert) [inline web-api/src/documents/generateStampedCoversheetLambda.js:16]: I would say that the proxy that is calling this endpoint should not send the docketNumber or docketEntryId over in the post body.  Right now I think there are two docketNumbers being passed in and it isn't clear without reading the code which one wins.

**Review by codyseibert — APPROVED** [review]

File: web-client/src/presenter/actions/getComputedFormDateFactoryAction.js [hunk 9]
```diff
@@ -60,6 +60,5 @@ export const getComputedFormDateFactoryAction = (
 
     return { [stateKey]: computedDate };
   };
-
   return computeFormDateAction;
 };
```
Comment (rosiehsmith) [inline web-client/src/presenter/actions/getComputedFormDateFactoryAction.js]: undo this

File: web-client/src/presenter/computeds/applyStampFormHelper.js [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+  const { isPdfAlreadyStamped, stampApplied, stampData } = pdfForSigning;
+
+  const CUSTOM_ORDER_MAX_LENGTH = 60;
+  const customOrderTextCharacterCount = customText?.length
+    ? CUSTOM_ORDER_MAX_LENGTH - customText?.length
+    : CUSTOM_ORDER_MAX_LENGTH;
+
+  const canSaveStampOrder =
+    !!form.disposition && get(state.pdfForSigning.stampApplied);
+  const cursorClass =
```
Comment (rosiehsmith) [inline web-client/src/presenter/computeds/applyStampFormHelper.js]: this could be removed

**Pull request merged**